src/mongodb.c

```diff
@@ -305,6 +305,45 @@ static int mg_process_database(
   return result;
 }
 
+// This code is identical to the method "mongoc_client_get_database_names"
+// in the Mongo driver, except that it doesn't filter out the database
+// called "local". This allows us to correctly enumerate and gather stats
+// from all databases when there is a "local" database (the normal case),
+// and when there is not (as in the sharding case).
+static char **
+mg_get_database_names (mongoc_client_t *client, bson_error_t *error)
+{
+   bson_iter_t iter;
+   const char *name;
+   char **ret = NULL;
+   int i = 0;
+   mongoc_cursor_t *cursor;
+   const bson_t *doc;
+
+   BSON_ASSERT (client);
+
+   cursor = mongoc_client_find_databases (client, error);
```

Comment:
Should we be checking the error here? I don't know much about these APIs, but it seems that the call in line 338 will overwrite the contents of error...